File: src/MediCore.Desktop/lib/features/prescriptions/domain/entities/prescription.dart
import 'prescription_item.dart';

class Prescription {
  final String id;
  final String encounterId;
  final String patientId;
  final String doctorId;
  final String prescriptionNumber;
  final String status;
  final String? notes;
  final List<PrescriptionItem> items;
  final DateTime createdAt;

  const Prescription({
    required this.id,
    required this.encounterId,
    required this.patientId,
    required this.doctorId,
    required this.prescriptionNumber,
    required this.status,
    this.notes,
    required this.items,
    required this.createdAt,
  });
}

File: src/MediCore.API/MediCore.Domain/Entities/Medical/Prescription.cs
using MediCore.Domain.Common;

namespace MediCore.Domain.Entities.Medical;

public class Prescription : BaseEntity
{
    public Guid TenantId { get; set; }
    public Guid EncounterId { get; set; }
    public Guid PatientId { get; set; }
    public Guid DoctorId { get; set; }
    public string PrescriptionNumber { get; set; } = string.Empty;
    public string Status { get; set; } = "draft";
    public string? Notes { get; set; }
    public DateTime? DispensedAt { get; set; }
    public Guid? DispensedBy { get; set; }

    // Navigation Properties
    public Encounter Encounter { get; set; } = null!;
    public Patient Patient { get; set; } = null!;
    public ICollection<PrescriptionItem> Items { get; set; } = new List<PrescriptionItem>();
}